hadoop-hdfs-project/hadoop-hdfs-native-client/src/main/native/libhdfs/jni_helper.c

```diff
@@ -534,6 +545,21 @@ JNIEnv* getJNIEnv(void)
 
     state->env = getGlobalJNIEnv();
     mutexUnlock(&jvmMutex);
+
+    if (!jclassesInitialized) {
+      mutexLock(&jclassInitMutex);
```

Comment:
I don't see a corresponding unlock for this

Reply:
Fixed